doc/DHCPv4_Gateway/DHCPv4_gateway.md

```diff
@@ -0,0 +1,133 @@
+# DHCPv4 - Specify gaaddr as Primary Interface's Gateway explicitly #
+
+## Table of Content 
+1. [Scope](#Scope)
+2. [Definitions](#Definitions/Abbreviations)
+3. [Overview](#Overview)
+4. [Requirements](#Requirements)
+5. [High Level Design](#High-Level-Design)
+6. [Test](#Testing-Considerations)
+
+### Revision  
+
+### Scope  
+
+This document describes the High Level Design of 'secondary' interfaces of vlan and use of primary (non-secondary) interface for dhcpv4 relay's gaaddr.
+
+### Definitions/Abbreviations 
+
+
+
+### Overview 
+
+A vlan can support multiple subnets/interfaces. Some of these interfaces (ideally all except one) can be marked as 'secondary'.
+Secondary subnets IPs are not assigned by dhcpv4. 
+
+When a dhcpv4 packet is forwarded via (isc) dhcprelayagent, it embeds its interface's gateway address as 'giaddr' for return communication.
+In case of multiple interfaces (one primary, multiple secondary), dhcprelayagent misconfigures this giaddr to any of the interfaces. 
+This change fixes this behaviour by explicitly specifying the primary interface/gateway.
+These secondary interfaces are attributed with 'secondary':'true' in config_db/yang/cli for VLAN_INTERFACE.
+
+### Requirements
+
+1. Support a new member 'secondary' of VLAN_INTERFACE in config_db.
+2. Support parsing and assignment of subnets from minigraph/json to config_db. 
+3. Support specifying non-secondary interfaces' gateway address to command line arguments to /usr/sbin/dhcrelay as -g.
+4. isc-dhcp/dhcrelay to support '-g gateway' argument, by porting an existing patch.
```

Comment:
Hope we handle any on-the-fly primary/secondary flag changes in CONFIG_DB for VLAN interface? i.e restarting the DHCP relay to pickup right primary IP address.

Reply:
Add relay address by cli would cause dhcp_relay container to restart https://github.com/sonic-net/sonic-buildimage/blob/c71fb3a30f3d384438f2e4861c80c28aea398fef/dockers/docker-dhcp-relay/cli/config/plugins/dhcp_relay.py#L77